kadmin/get.c

```diff
@@ -500,6 +503,212 @@ do_get_entry(krb5_principal principal, void *data)
     return ret;
 }
 
+static int
+tl_data2json(kadm5_principal_ent_rec *p, struct get_entry_data *e, heim_dict_t d)
+{
+    krb5_tl_data *tl;
+    heim_object_t o = NULL;
+
+    for (tl = p->tl_data; tl; tl = tl->tl_data_next) {
+        switch (tl->tl_data_type) {
+        case KRB5_TL_PASSWORD:
+            if (e->options->json_with_keys_flag)
+                heim_dict_set_value(d, HSTR("password"),
+                                    o = heim_string_create(tl->tl_data_contents));
+            else
+                heim_dict_set_value(d, HSTR("password"),
+                                    heim_bool_create(1));
+            heim_release(o);
+            continue;
+        default:
+            /* XXX Add more! */
+            continue;
+        }
+    }
+    return 0;
+}
+
+static void
+keys2json(kadm5_principal_ent_rec *p, struct get_entry_data *e, heim_dict_t d)
+{
+    heim_array_t a = heim_array_create();
```

Comment:
I mean, this could return NULL, but... realistically for a tool it's probably fine to assume malloc success.

Reply:
Yeah, I should still not deref NULL in tools.